4.0/zh-cn/0x10-V1-Architecture.md

```diff
@@ -0,0 +1,142 @@
+# V1 架构、设计和威胁建模
+
+## 控制目标
+
+在许多组织中，安全架构几乎已成为一门失传的艺术。 在DevSecOps时代，企业架构师的日子已经过去。应用安全领域必须迎头赶上，采用敏捷安全原则，同时将领先的安全架构原则重新介绍给软件从业者。 A架构不是一种实施，而是一种思考问题的方式，它可能有许多不同的答案，而没有一个单一的“正确”答案。 很多时候，安全被视为不灵活的，要求开发人员以特定方式修复代码，而开发人员可能知道解决问题的更好方法。 对于架构来说，没有单一的、简单的解决方案，尝试寻找这种方案，是对软件工程领域的一种损害。
+
+一个Web应用程序的具体实现，很可能在其生命周期中不断被修改，但整体架构可能很少改变，而是缓慢发展。 安全架构也是一样的，身份验证——我们今天需要、明天需要、五年后也需要。 如果我们今天做出合理的决定，选择和复用符合架构的解决方案，那么就可以节省大量的精力、时间和金钱。 例如，十年前，多因素认证很少被实施。
+
+如果开发人员已经在单一的“安全标识提供程序模型”上有所投入（例如SAML联邦认证），身份提供者可以更新以纳入新的要求（例如NIST 800-63标准），同时不改变原始应用程序的接口。 如果许多应用程序共享相同的安全架构和组件，那么它们都将同时从这次升级中受益。 然而，SAML 并不总是最好或最合适的身份验证解决方案——随着需求的变化，可能需要替换为其他解决方案。 像这样的更改要么很复杂，成本高到需要完全重写，要么在没有安全架构的情况下完全不可能。
+
+在本章中，ASVS涵盖了任何良好安全架构的主要方面：可用性、保密性、完整性、不可抵赖性和隐私。 这些安全原则中的每一条，都必须适用并内置于所有应用程序中。 “左移”至关重要，从安全编码Checklists、指导和培训、编码和测试、构建、部署、配置和操作开始，到后续的独立测试，确保所有安全控制存在且功能正常。 这最后一步，曾经是我们作为一个行业所做的一切，但当开发人员每天数十次或数百次地推送代码时，就已经不够了。 应用安全专业人员必须跟上敏捷技术的步伐，这意味着要适应开发人员的工具，学习编码，并与开发人员一起工作，而不是在其他人离开的几个月后再批评项目。
+
+## V1.1 安全软件开发生命周期
+
+| # | 说明 | L1 | L2 | L3 | CWE |
+| :---: | :--- | :---: | :---:| :---: | :---: |
+| **1.1.1** | 验证使用安全的软件开发生命周期，在开发的各个阶段解决安全问题。 ([C1](https://owasp.org/www-project-proactive-controls/#div-numbering)) | | ✓ | ✓ | |
+| **1.1.2** | 验证在每次设计变更或sprint计划中使用威胁建模，以识别威胁、计划对策、促进适当的风险响应，并指导安全测试。 | | ✓ | ✓ | 1053 |
+| **1.1.3** | 验证所有用户信息和功能是否包含功能安全约束，例如 “作为一个用户，我应该能够查看和编辑我的个人资料。我不应该能够查看或编辑其他人的资料” | | ✓ | ✓ 1110 |
+ **1.1.4** | 验证应用程序所有的信任边界、组件和重要数据流的文档，判断其合理性。| | ✓ | ✓ | 1059 |
+| **1.1.5** | 验证应用程序的高级架构及远程连接服务涉及的定义和安全分析。 ([C1](https://owasp.org/www-project-proactive-controls/#div-numbering)) | | ✓ | ✓ | 1059 |
+| **1.1.6**| 验证集中、简单（设计）、安全、经过审查、和可重复使用的安全控制措施的实施情况，以避免重复、缺失、无效或不安全的控制措施。（[C10](https://owasp.org/www-project-proactive-controls/#div-编号）| | ✓ | ✓ | 637 |
```

Comment:
incorrect link, C10

Reply:
solved in https://github.com/hi-unc1e/ASVS/commit/bc2853e120cd43416252632d097dbff389c54577